flang/lib/Lower/OpenMP.cpp

```diff
@@ -716,6 +727,62 @@ genOMP(Fortran::lower::AbstractConverter &converter,
   createBodyOfOp<omp::CriticalOp>(criticalOp, converter, currentLocation, eval);
 }
 
+ void Fortran::lower::genOpenMPSectionsBlock(Fortran::lower::AbstractConverter &converter, Fortran::lower::pft::Evaluation &eval){
+     auto &firOpBuilder = converter.getFirOpBuilder(); 
+     auto currentLocation = converter.getCurrentLocation();
+     mlir::omp::SectionOp sectionOp = firOpBuilder.create<mlir::omp::SectionOp>(currentLocation);
+     createBodyOfOp<omp::SectionOp>(sectionOp, converter, currentLocation, eval);
+ }
+
+ // TODO: Add support for reduction
+ static void
+ genOMP(Fortran::lower::AbstractConverter &converter,
+        Fortran::lower::pft::Evaluation &eval,
+        const Fortran::parser::OpenMPSectionsConstruct &sectionsConstruct) {
+     auto &firOpBuilder = converter.getFirOpBuilder();
+     auto currentLocation = converter.getCurrentLocation();
+     SmallVector<Value> privateClauseOperands, firstPrivateClauseOperands,
+         lastPrivateClauseOperands, reductionVars, allocateOperands,
+         allocatorOperands;
+     mlir::Attribute noWaitClauseOperand;
+     const auto &sectionsClauseList = std::get<Fortran::parser::OmpClauseList>(
+             std::get<Fortran::parser::OmpBeginSectionsDirective>(sectionsConstruct.t).t);
+
+     for(const auto &clause : sectionsClauseList.v){
+
+         if (const auto &lastPrivateClause = std::get_if<Fortran::parser::OmpClause::Lastprivate>(&clause.u)){
+             const Fortran::parser::OmpObjectList &ompObjectList = lastPrivateClause->v;
+             genObjectList(ompObjectList, converter, lastPrivateClauseOperands);
+         }
+
+         if (const auto &allocateClause =
+                 std::get_if<Fortran::parser::OmpClause::Allocate>(&clause.u)) {
+           handleAllocateClause(converter, allocateClause->v, allocatorOperands,
+                                allocateOperands);
+         }
+     }
+     mlir::omp::SectionsOp sectionsOp =
+         firOpBuilder.create<mlir::omp::SectionsOp>(
+             currentLocation, privateClauseOperands, firstPrivateClauseOperands,
+             lastPrivateClauseOperands, reductionVars, nullptr,
```

Comment:
If possible use llvm::None here, it seems that there is a [constructor](https://llvm.org/doxygen/Optional_8h_source.html#l00245) that takes that argument.

Reply:
Ah okay, it looks like the optional attributes do not use llvm::Optional for builders. Should be okay then.